lib/daru/dataframe.rb

```diff
@@ -1270,13 +1270,66 @@ def tail quantity=10
 
     alias :last :tail
 
-    # Returns a vector with sum of all vectors specified in the argument.
-    # If vecs parameter is empty, sum all numeric vector.
-    def vector_sum vecs=nil
+    # Sum all numeric/specified vectors in the DataFrame.
+    #
+    # Returns a new vector that's a containing a sum of all numeric
+    # or specified vectors of the DataFrame. By default, if the vector
+    # contains a nil, the sum is nil.
+    # With :skipnil argument set to true, nil values are assumed to be
+    # 0 (zero) and the sum vector is returned.
+    #
+    # @param opts [Array] List of vectors to sum. Default is nil in which case
+    #   all numeric vectors are summed.
+    #
+    # @param opts [Boolean] :skipnil Consider nils as 0. Default is false.
+    #
+    # @return Vector with sum of all vectors specified in the argument.
+    #   If vecs parameter is empty, sum all numeric vector.
+    #
+    # @example
+    #    df = Daru::DataFrame.new({
+    #       a: [1, 2, nil],
+    #       b: [2, 1, 3],
+    #       c: [1, 1, 1]
+    #     })
+    #    => #<Daru::DataFrame(3x3)>
+    #           a   b   c
+    #       0   1   2   1
+    #       1   2   1   1
+    #       2 nil   3   1
+    #    df.vector_sum [:a, :c]
+    #    => #<Daru::Vector(3)>
+    #       0   2
+    #       1   3
+    #       2 nil
+    #    df.vector_sum
+    #    => #<Daru::Vector(3)>
+    #       0   4
+    #       1   4
+    #       2 nil
+    #    df.vector_sum skipnil: true
+    #    => #<Daru::Vector(3)>
+    #           c
+    #       0   4
+    #       1   4
+    #       2   4
+    #
+    def vector_sum(*args)
+      defaults = {vecs: nil, skipnil: false}
```

Comment:
What's the advantage of this over just having named args with defaults since there's only 2?

Reply:
I believe the advantage is you can do `vector_sum(skip_nil: true)` instead of `vector_sum(nil, true)` of your PR, which is a bit inelegant. 

In addition, when we'll drop Ruby 2.0 support (which is immediately after Google Summer of Code end), here would be proper keyword args. So, I believe it is a good solution.

Reply:
@baarkerlounger The intent was to provide an easy way skip the first argument while providing the second one. At the cost of some arg processing complexity, arguably the user interaction gets simplified a bit. The following combinations work.

```ruby
df.vector_sum
df.vector_sum skipnil: true
df.vector_sum [:a, :b], skipnil: true
df.vector_sum [:a, :b]
```